packages/did/src/DidDetails/LightDidDetails.ts

```diff
@@ -8,245 +8,163 @@
   }
 
-  /**
-   * Create a new instance of [[LightDidDetails]] from the provided details.
-   * Private keys are assumed to already live in another storage, as it contains reference only to public keys.
-   *
-   * @param details The DID creation details.
-   * @param details.authenticationKey The light DID authentication key.
-   * @param details.encryptionKey The optional light DID encryption key.
-   * @param details.serviceEndpoints The optional light DID service endpoints.
-   *
-   * @returns The resulting [[LightDidDetails]].
-   */
-  public static fromDetails({
-    authenticationKey,
-    encryptionKey = undefined,
-    serviceEndpoints = [],
-  }: LightDidCreationDetails): LightDidDetails {
-    checkLightDidCreationDetails({
-      authenticationKey,
-      encryptionKey,
-      serviceEndpoints,
-    })
-    const encodedDetails = serializeAndEncodeAdditionalLightDidDetails({
-      encryptionKey,
-      serviceEndpoints,
-    })
-    // Validity is checked in checkLightDidCreationDetails
-    const authenticationKeyTypeEncoding = getEncodingForVerificationKeyType(
-      authenticationKey.type
-    ) as string
-
-    // A KILT light DID identifier becomes <key_type_encoding><kilt_address>
-    const id = authenticationKeyTypeEncoding.concat(
-      encodeAddress(authenticationKey.publicKey, ss58Format)
-    )
-
-    let uri = getKiltDidFromIdentifier(id, 'light', LIGHT_DID_LATEST_VERSION)
-    if (encodedDetails) {
-      uri = uri.concat(':', encodedDetails) as DidUri
-    }
-
-    // Authentication key always has the #authentication ID.
-    const keys: PublicKeys = {
-      [authenticationKeyId]: { ...authenticationKey },
-    }
-    const keyRelationships: MapKeysToRelationship = {
-      authentication: new Set([authenticationKeyId]),
-    }
+  if (encryptionKey) {
+    details.keyAgreement = [
+      {
+        id: encryptionKeyId, // Encryption key always has the #encryption ID.
+        ...encryptionKey,
+      },
+    ]
+  }
 
-    // Encryption key always has the #encryption ID.
-    if (encryptionKey) {
-      keys[encryptionKeyId] = encryptionKey
-      keyRelationships.keyAgreement = new Set([encryptionKeyId])
-    }
+  return details
+}
 
-    const endpoints = serviceEndpoints.reduce((res, service) => {
-      res[service.id] = service
-      return res
-    }, {})
+/**
+ * Create a new instance of [[LightDidDetails]] by parsing the provided input URI.
+ * This is possible because of the self-describing and self-containing nature of light DIDs.
+ * Private keys are assumed to already live in another storage, as it contains reference only to public keys.
+ *
+ * @param uri The DID URI to parse.
+ * @param failIfFragmentPresent Whether to fail when parsing the URI in case a fragment is present or not, which is not relevant to the creation of the DID. It defaults to true.
+ *
+ * @returns The resulting [[LightDidDetails]].
+ */
+export function fromUri(uri: DidUri, failIfFragmentPresent = true): DidDetails {
```

Comment:
could also be renamed to make sure it still expresses what it creates

Reply:
I think if we go this route we need to work on the constructors. We have a function in FullDidDetails that does the exact same thing for on-chain DIDs but has a different name - we really need to think about whether we want to integrate those two, and if not work on the naming.

Reply:
talked about it 1 on 1, this is likely going to be `parseDetailsFromLightDid`, then we'll have a `queryDid` or `queryFullDid`. We could think about a higher level function (`resolve`?) that accepts either and chooses the correct path (parse or query) depending on the did type

Reply:
As discussed, we’ll have `createDetails`, `parseDetailsFromLightDid`, and `query`. 00d7c08f and 8ce24621